drivers/dma/dma_silabs_siwx917.c

```diff
@@ -9,20 +9,23 @@
 #include <zephyr/sys/util.h>
 #include <zephyr/device.h>
 #include <zephyr/drivers/dma.h>
+#include <zephyr/sys/mem_blocks.h>
 #include <zephyr/drivers/clock_control.h>
 #include <zephyr/logging/log.h>
 #include <zephyr/sys/sys_io.h>
 #include <zephyr/types.h>
 #include "rsi_rom_udma_wrapper.h"
+#include "rsi_rom_udma.h"
 #include "rsi_udma.h"
 #include "sl_status.h"
 
-#define DT_DRV_COMPAT          silabs_siwx917_dma
-#define DMA_MAX_TRANSFER_COUNT 1024
-#define DMA_CH_PRIORITY_HIGH   1
-#define DMA_CH_PRIORITY_LOW    0
-#define VALID_BURST_LENGTH     0
-#define UDMA_ADDR_INC_NONE     0X03
+#define DT_DRV_COMPAT                    silabs_siwx917_dma
+#define DMA_MAX_TRANSFER_COUNT           1024
+#define DMA_CH_PRIORITY_HIGH             1
+#define DMA_CH_PRIORITY_LOW              0
+#define VALID_BURST_LENGTH               0
+#define UDMA_ADDR_INC_NONE               0X03
+#define UDMA_MODE_PER_ALT_SCATTER_GATHER 0x07
```

Comment:
Since you're modifying the `UDMA_ADDR_INC_NONE` line, please fix the uppercase `X` to lowercase for consistency.